ToolBox/GDAXDownloader/Program.cs

```diff
@@ -0,0 +1,75 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+using System;
+using System.Globalization;
+using QuantConnect.Configuration;
+using QuantConnect.Logging;
+
+namespace QuantConnect.ToolBox.GDAXDownloader
+{
+    class Program
+    {
+        /// <summary>
+        /// GDAX Downloader Toolbox Project For LEAN Algorithmic Trading Engine.
+        /// </summary>
+        static void Main(string[] args)
+        {
+            if (args.Length == 2)
+            {
+                args = new [] { args[0], DateTime.UtcNow.ToString("yyyyMMdd"), args[1] };
+            }
+            else if (args.Length < 3)
+            {
+                Console.WriteLine("Usage: GDAX Downloader SYMBOL FROMDATE TODATE");
+                Console.WriteLine("FROMDATE = yyyymmdd");
+                Console.WriteLine("TODATE = yyyymmdd");
+                Environment.Exit(1);
+            }
+
+            try
+            {
+                // Load settings from command line
+                var startDate = DateTime.ParseExact(args[1], "yyyyMMdd", CultureInfo.InvariantCulture);
+                var endDate = DateTime.ParseExact(args[2], "yyyyMMdd", CultureInfo.InvariantCulture);
+
+                // Load settings from config.json
+                var dataDirectory = Config.Get("data-directory", "../../../Data");
+                //todo: will download any exchange but always save as gdax
+                // Create an instance of the downloader
+                const string market = Market.GDAX;
+                var downloader = new GDAXDownloader();
+
+                // Download the data
+                var symbolObject = Symbol.Create(args[0], SecurityType.Crypto, market);
+                var data = downloader.Get(symbolObject, Resolution.Hour, startDate, endDate);
+
+                // Save the data
+                
+                var writer = new LeanDataWriter(Resolution.Hour, symbolObject, dataDirectory, TickType.Quote);

```

Comment:
You are correct. Nothing is preventing us from using options for different Resolutions. This was like version 1. Thanks

Reply:
While I agree this may be version 1, the ability to down _only_ hourly data is not very useful to the rest of the community. Hourly data is probably the _least_ often resolution used. The other downloader programs accept a resolution as a command line argument, I'd recommend scooping the code from there.